### PR TITLE
Allow for name overrides of resources

### PR DESCRIPTION
**Pull request by jaumann**
- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Comment by jmlrt** [comment]
jenkins test this please

File: elasticsearch/templates/service.yaml [hunk 1]
```diff
@@ -2,7 +2,15 @@
 kind: Service
 apiVersion: v1
 metadata:
+<<<<<<< HEAD
```
Comment (jmlrt) [inline elasticsearch/templates/service.yaml]: It seems that you committed a merge conflict by mistake

Comment (jaumann) [reply elasticsearch/templates/service.yaml]: Oops! Fixed. Thanks!

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Pull request merged**